### PR TITLE
Add a sub menu item for NagVis underneath "Maps"

### PR DESCRIPTION
**Pull request by dnsmichi**
This allows other modules to use the "Maps" menu structure as well.
I'll prepare PRs for the "map" and "globe" module.

### CODE REVIEW

File: configuration.php [hunk 1]
```diff
@@ -4,6 +4,8 @@
 use Icinga\Module\Nagvis\RestrictionHelper;
 
 $section = $this->menuSection(N_('Maps'))
+    ->setIcon('globe')
+    ->add($this->translate('NagVis'))
```
Comment (Thomas-Gelf) [inline configuration.php]: Please note that you are switching context here. After this, $section is no longer the main menu entry but a sub-menu. Following code makes then wrong assumptions and creates sub-sub-menus. While in theory we support such, there are no explicitly specified rendering rules for them. So this change has the potential to break things.

**Comment by Thomas-Gelf** [comment]
@dnsmichi: while I absolutely second the attempt to generalize this, please have a look at my inline comment and double-check this. NagVis might have a lot of submenu entries. Also, the change introduces an icon for a submenu. This is supported in the code but might look weird as it isn't used anywhere else.

**Comment by Thomas-Gelf** [comment]
@dnsmichi: eventually consult @lippserd on this ;)

**Comment by dnsmichi** [comment]
Mh in terms of the `$section` variable, I agree, this should be on the same level as NagVis itself. Yet it seems that the code depends on config snippets being stored somewhere in the module's menu.ini

If I want to add a NagVis map to the menu, I cannot do that beneath dynamically added sections, but only for those which are configured (behaves the same with and without my patch).

screen shot 2017-08-04 at 15 32 25


In terms of icons for sub menu items - I've done that via menu.ini entries and it worked fine. Anything specific you're referring to?

```
[Graphing]
type = "menu-item"
target = "_main"
url = "#"
icon = "chart-bar"

[Grafana]
type = "menu-item"
target = "_main"
url = "/icingaweb2/iframe?url=http://192.168.33.5:8004/dashboard/db/icinga2"
icon = "chart-bar"
parent = "Graphing"
 ```

**Comment by lippserd** [comment]
I guess what Thomas is trying to say is that it may look weird if sub menu entries have an icon because no other sub menu entry we use has one.

**Comment by dnsmichi** [comment]
Discussed it offline with @lippserd and we agreed on the section fix, and to drop the icon. I wasn't aware that you normally don't use sub menu icons. I found it more beautiful when creating the vagrant box demo :)

**Comment by Hipska** [comment]
Hi, when will a new version with this feature be released?

See also https://monitoring-portal.org/t/nagvis-module-creates-duplicate-maps-menu/307

**Pull request merged**